sympy/printing/smtlib.py

```diff
@@ -394,6 +481,41 @@ def smtlib_code(
     ])
 
 
+def _contains_custom_predicate(expr):
+    custom_predicate = [Q.integer, Q.even, Q.odd, Q.prime, Q.composite]
+    if isinstance(expr, AppliedPredicate) and expr.function in custom_predicate:
+        return True
+    elif isinstance(expr, (And, Or, Not)):
+        return any([_contains_custom_predicate(arg) for arg in expr.args])
```

Comment:
using `any([ ... ])` forces the list comprehension to solve into a list. Maybe it's more efficient to just remove the square brackets?

Reply:
`return any(_contains_custom_predicate(arg) for arg in expr.args)`